libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -679,40 +629,32 @@ export class BotFrameworkAdapter extends BotAdapter {
      * @param context Context for the current turn of conversation with the user.
      * @param activity New activity to replace a current activity with.
      */
-    public updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<void> {
-        try {
-            if (!activity.serviceUrl) { throw new Error(`BotFrameworkAdapter.updateActivity(): missing serviceUrl`); }
-            if (!activity.conversation || !activity.conversation.id) {
-                throw new Error(`BotFrameworkAdapter.updateActivity(): missing conversation or conversation.id`);
-            }
-            if (!activity.id) { throw new Error(`BotFrameworkAdapter.updateActivity(): missing activity.id`); }
-            const client: ConnectorClient = this.createConnectorClient(activity.serviceUrl);
-
-            return client.conversations.updateActivity(
-                activity.conversation.id,
-                activity.id,
-                activity as Activity
-            ).then(() => {
-                // noop
-            });
-        } catch (err) {
-            return Promise.reject(err);
+    public async updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<void> {
+        if (!activity.serviceUrl) { throw new Error(`BotFrameworkAdapter.updateActivity(): missing serviceUrl`); }
+        if (!activity.conversation || !activity.conversation.id) {
+            throw new Error(`BotFrameworkAdapter.updateActivity(): missing conversation or conversation.id`);
         }
+        if (!activity.id) { throw new Error(`BotFrameworkAdapter.updateActivity(): missing activity.id`); }
+        const client: ConnectorClient = this.createConnectorClient(activity.serviceUrl);
+        await client.conversations.updateActivity(
```

Comment:
I don't understand why the updateActivity here wouldn't just take an Activity. That would allow all the valiations to be done in there, and it seems quite a bit cleaner. 

It's quite odd that you're passing in the id's off the activity AND the activity itself. Strongly consider refactoring to cleanup the implementation. 

There is no test coverage for a circumstance for where the activity.id and the activity don't match. Is that a valid scenario? If so, we'll have to update the C# code.

Reply:
This is from the swagger generated code. It takes an ID because the REST path generated has an id.

---
In reply to: [225354928](https://github.com/Microsoft/botbuilder-js/pull/544#discussion_r225354928) [](ancestors = 225354928)